### ISSUE TITLE
Check compatibility based on Kubernetes cluster version before installing

### ISSUE THREAD

**Issue by christian-kreuzberger-dtx** [keptn/keptn#1326]
**Story**: As a user I want to install Keptn only on a Kubernetes Cluster that is compatible with Keptn.

## Technical Details
When running `keptn install` the command (or the installer) should check if the cluster version is compatible with the current Keptn version (e.g., Keptn 0.6.0 is compatible with Kubernetes 1.13, 1.14 and 1.15, but not 1.16 AFAIK).

We could implement this check by executing
```console
kubectl version
```
and parsing the response of the server version
```
Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.11-gke.14", GitCommit:"56d89863d1033f9668ddd6e1c1aea81cd846ef88", GitTreeState:"clean", BuildDate:"2019-11-07T19:12:22Z", GoVersion:"go1.12.11b4", Compiler:"gc", Platform:"linux/amd64"}
```

basically we need to check for `Major == "1"` and minor version `>= '13'` and `<= '15' (it's a string, so not sure whether we can easily check that with the `>=`/`<=` operators)

If we detect an unsupported Kubernetes version, the installer should exit with an appropriate error message.

## Definition of Done

When executing `keptn install ...`, the installer should not start execution if the cluster version is incompatible.
(Note: Whether this is done by a check in the CLI or in the installer needs to be discussed)

**Comment by SVilgelm** [comment]
Hi @christian-kreuzberger-dtx,
In order to get to know this project, i decided to solve this task. I have prepared a simple PR: #1382
To solve the problem with comparing the versions I use the https://github.com/hashicorp/go-version (Mozilla Public License 2.0 license)
I think the best place to have this validation is in the CLI, since it already checks that the `kubectl` exists

**Comment by SVilgelm** [comment]
I have one open question - what is a best way to pass the constraints? I temporary put them into a constant, but I think I should come from the manifests

**Comment by SVilgelm** [comment]
Also I will add the unit tests if you are Ok with my solution

**Comment by christian-kreuzberger-dtx** [comment]
Hi @SVilgelm and thank you for your contribution!

* Using `hashicorp/go-version` is fine, this looks exactly like what we need. I believe the license used should be fine.
* Validation in the CLI is fine for now, though long term we will need the same validation logic in another place, but that's something we will figure out in the future.
* "what is a best way to pass the constraints?" - this is a very good question, thanks for raising that.

At the moment we do not have a central place to put such variables. Instead, we went the following approach:

When building the CLI we add a build variable, see
    https://github.com/keptn/keptn/blob/ad7cddda5401c9d447749bb953f2a0cbf66e1b12/cli/buildCLI.sh#L4-L7
    and
    https://github.com/keptn/keptn/blob/ad7cddda5401c9d447749bb953f2a0cbf66e1b12/travis-scripts/build_cli.sh#L8 (as well as Lines 18 and 28)
We then set the environment variable using travis-ci, e.g.: https://github.com/keptn/keptn/blob/ad7cddda5401c9d447749bb953f2a0cbf66e1b12/.travis.yml#L516

I think this approach is good for now, and it is flexible enough such that we can remove it if we ever need.
